Cubical/Structures/Group/Morphism.agda

```diff
@@ -0,0 +1,253 @@
+{-# OPTIONS --cubical --no-import-sorts --safe #-}
+
+module Cubical.Structures.Group.Morphism where
+
+open import Cubical.Foundations.Prelude
+open import Cubical.Foundations.Equiv
+open import Cubical.Foundations.Equiv.HalfAdjoint
+open import Cubical.Foundations.HLevels
+open import Cubical.Foundations.Isomorphism
+open import Cubical.Foundations.Univalence
+open import Cubical.Foundations.Transport
+open import Cubical.Foundations.SIP renaming (SNS-PathP to SNS)
+open import Cubical.Foundations.Function using (_∘_)
+open import Cubical.Foundations.GroupoidLaws
+
+open import Cubical.Structures.Group.Base
+open import Cubical.Structures.Macro
+open import Cubical.Structures.NAryOp
+open import Cubical.Structures.Pointed
+open import Cubical.Structures.Semigroup hiding (⟨_⟩)
+open import Cubical.Structures.Monoid hiding (⟨_⟩)
+
+open import Cubical.Functions.Embedding
+open import Cubical.Data.Sigma
+open Iso
+
+private
+  variable
+    ℓ ℓ' ℓ'' : Level
+
+-- The following definition of GroupHom and GroupIso are level-wise heterogeneous.
+-- This allows for example to deduce that G ≡ F from a chain of isomorphisms
+-- G ≃ H ≃ F, even if H does not lie in the same level as G and F.
+
+isGroupHom : (G : Group {ℓ}) (H : Group {ℓ'}) (f : ⟨ G ⟩ → ⟨ H ⟩) → Type _
+isGroupHom G H f = (x y : ⟨ G ⟩) → f (x G.+ y) ≡ (f x H.+ f y) where
+  module G = Group G
+  module H = Group H
+  
+isPropIsGroupHom : (G : Group {ℓ}) (H : Group {ℓ'}) {f : ⟨ G ⟩ → ⟨ H ⟩} → isProp (isGroupHom G H f)
+isPropIsGroupHom G H {f} = isPropΠ2 λ a b → Group.is-set H _ _
+
+record GroupHom (G : Group {ℓ}) (H : Group {ℓ'}) : Type (ℓ-max ℓ ℓ') where
+  constructor grouphom
+
+  field
+    fun : ⟨ G ⟩ → ⟨ H ⟩
+    isHom : isGroupHom G H fun
+
+record GroupIso (G : Group {ℓ}) (H : Group {ℓ'}) : Type (ℓ-max ℓ ℓ') where
```

Comment:
@ecavallo Didn't you call this GroupEquiv?

Reply:
Oops, I forgot to add a few files before committing the merge.